src/nupic/utils/GroupBy.hpp

```diff
@@ -183,12 +182,29 @@ namespace nupic
   // ==========================================================================
 
   template<typename Iterator0, typename KeyFn0,
+           typename KeyType = typename std::result_of<
+             KeyFn0(decltype(*std::declval<Iterator0>()))
+               >::type>
+  static KeyType min_front_key(KeyType frontrunner,
```

Comment:
I'm not super keen on ignoring naming conventions here. Unless you want to propose some specific change to our naming guidelines.